static/src/javascripts/projects/common/modules/tailor/fetch-data.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+
+import fetchJson from 'lib/fetch-json';
+import { getCookie } from 'lib/cookies';
+import { local as localStorage } from 'lib/storage';
+import reportError from 'lib/report-error';
+import config from 'lib/config';
+
+const BROWSER_ID = getCookie('bwid');
+const URLS = {
+    suggestions: 'https://tailor.guardianapis.com/suggestions?browserId=',
+};
+
+const getURL = (type: string, queryParams?: Object): ?string => {
+    if (!BROWSER_ID || !URLS[type]) {
+        return;
+    }
+
+    let baseURL = URLS[type] + BROWSER_ID;
+
+    if (queryParams) {
+        Object.keys(queryParams).forEach(key => {
+            if (queryParams) {
```

Comment:
Flow is [pessimistic about refinements](https://github.com/facebook/flow/issues/2986#issuecomment-265952300), so we'll have to live with this for now. Perhaps refactoring into a `for ... in` would help as it would avoid creating the additional function scope.